### DISCUSSION TITLE
What does the `engine` metadata do?

### DISCUSSION THREAD

**Discussion by iMacTia** [rubyatscale/packs-rails#14] [Q&A] [answered]
The README says that you can [make your package an engine](https://github.com/rubyatscale/stimpack#making-your-package-an-engine), but it's unclear what difference that actually makes.
Could this be clarified and/or documented?

**Comment by ngan** [comment, marked as answer]
Hi @iMacTia -- This feature will promote your pack into a Rails Engine. It basically [creates](https://github.com/rubyatscale/stimpack/blob/main/lib/stimpack/pack.rb#L36-L40) (dynamically / automatically) a namespace for your pack and [adds namespace isolation](https://github.com/rubyatscale/stimpack/blob/main/lib/stimpack/stim.rb#L10-L12).

You can do that the nifty things you normally get with an Engine... eg [`mount` its routes](https://guides.rubyonrails.org/engines.html#mounting-the-engine), [scoped url helpers](https://guides.rubyonrails.org/engines.html#routes), or table name prefix.

**Reply by iMacTia** [reply]
Interesting! Thank you for clarifying 🙏 !

**Reply by exterm** [reply]
Ah. This was quite confusing to me since I'm used to using engines without namespace isolation. I'll put up a PR to add details to the README.
